pkg/sql/schemachanger/scbuild/view.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package scbuild
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/resolver"
+	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgcode"
+	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgerror"
+	"github.com/cockroachdb/cockroach/pkg/sql/privilege"
+	"github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/sqlerrors"
+	"github.com/cockroachdb/errors"
+)
+
+// dropDependentView goes over the dependencies in a give view and validates
+// that they can be dropped for cascade behavior.
+func (b *buildContext) dropDependentView(ctx context.Context, table catalog.TableDescriptor) {
+	// If this object is already in a dropped state,
+	// then nothing needs to be done for this dependency.
+	if table.Dropped() {
+		return
+	}
+	err := table.ForeachDependedOnBy(func(dep *descpb.TableDescriptor_Reference) error {
+		dependentDesc, err := b.Descs.GetImmutableTableByID(ctx, b.EvalCtx.Txn, dep.ID, tree.ObjectLookupFlagsWithRequired())
+		if err != nil {
+			panic(err)
+		}
+		err = b.AuthAccessor.CheckPrivilege(ctx, dependentDesc, privilege.DROP)
+		if err != nil {
+			panic(err)
+		}
+		b.dropDependentView(ctx, dependentDesc)
+		return nil
+	})
+	if err != nil {
+		panic(err)
+	}
+	viewNode := &scpb.View{
+		TableID:      table.GetID(),
+		DependedOnBy: make([]descpb.ID, 0, len(table.TableDesc().DependedOnBy)),
```

Comment:
We should take every opportunity to beef up the `catalog.TableDescriptor` interface if it's inadequate, to avoid `.TableDesc()` punting in new code.